Framework/DataObjects/src/EventList.cpp

```diff
@@ -3937,6 +3956,31 @@ void EventList::filterInPlace(Kernel::SplittingIntervalVec &splitter) {
   }
 }
 
+/**
+ * Initialize the detector ID's and event type of the destination event lists when splitting this list.
+ * @param partials : resulting partial lists of events after splitting's done
+ */
+void EventList::initializePartials(std::map<int, EventList *> partials) const {
```

Comment:
`EventList` is-an `ISpectrum` so some of this can be replaced with [`ISpectrum::copyInfoFrom(other)`](https://github.com/mantidproject/mantid/blob/main/Framework/API/src/ISpectrum.cpp#L24-L29) to reduce the amount of code here.